scripts/linters/pylint_extensions.py

```diff
@@ -1055,46 +1055,24 @@ class SingleSpaceAfterYieldChecker(checkers.BaseChecker):
         ),
     }
 
-    def process_module(self, node):
-        """Process a module to ensure that yield keywords are followed by
-        exactly one space, so matching 'yield *' where * is not a
+    def visit_yield(self, node):
+        """Visit every yield statement to ensure that yield keywords are
+        followed by exactly one space, so matching 'yield *' where * is not a
         whitespace character. Note that 'yield' is also acceptable in
         cases where the user wants to yield nothing.
 
         Args:
-            node: astroid.scoped_nodes.Function. Node to access module
+            node: astroid.nodes.Yield. Nodes to access yield statements.
                 content.
         """
-        in_multi_line_comment = False
-        multi_line_indicator = b'"""'
-        file_content = read_from_node(node)
-        for (line_num, line) in enumerate(file_content):
-            bare_line = line.strip()
-
-            # Single multi-line comment, ignore it.
-            if bare_line.count(multi_line_indicator) == 2:
-                continue
-
-            # Flip multi-line boolean depending on whether or not we see
-            # the multi-line indicator. Possible for multiline comment to
-            # be somewhere other than the start of a line (e.g. func arg),
-            # so we can't look at start of or end of a line, which is why
-            # the case where two indicators in a single line is handled
-            # separately (i.e. one line comment with multi-line strings).
-            if multi_line_indicator in bare_line:
-                in_multi_line_comment = not in_multi_line_comment
-
-            # Ignore anything inside a multi-line comment.
-            if in_multi_line_comment:
-                continue
-
-            # Whitespace to right of yield keyword is important for regex.
-            # Allows alphabet characters and underscore for cases where 'yield'
-            # is used at the start of a variable name.
-            source_line = line.lstrip()
-            if (source_line.startswith(b'yield') and
-                    not re.search(br'^(yield)( \S|$|\w)', source_line)):
-                self.add_message('single-space-after-yield', line=line_num + 1)
+        python_utils.PRINT(node)
+        line_number = node.fromlineno
+        line = linecache.getline(node.root().file, line_number).lstrip()
+        python_utils.PRINT(line_number)
+        python_utils.PRINT(line)
+        if (line.startswith(b'yield') and
+                not re.search(br'^(yield)( \S|$|\w)', line)):
```

Comment:
Can you please explain the regex used here?

Reply:
Here we are checking if the yield is at the beginning and then a space after that and then either no word after that or some nonspace variables.